core/workspace.js

```diff
@@ -320,31 +321,39 @@ Blockly.Workspace.prototype.deleteVariable = function(name) {
   }
 
   var workspace = this;
-  function doDeletion() {
-    Blockly.Events.setGroup(true);
-    for (var i = 0; i < uses.length; i++) {
-      uses[i].dispose(true, false);
-    }
-    Blockly.Events.setGroup(false);
-    workspace.variableList.splice(variableIndex, 1);
-  }
   if (uses.length > 1) {
     // Confirm before deleting multiple blocks.
     Blockly.confirm(
         Blockly.Msg.DELETE_VARIABLE_CONFIRMATION.replace('%1', uses.length).
         replace('%2', name),
         function(ok) {
           if (ok) {
-            doDeletion();
+            workspace.deleteVariableInternal_(name);
           }
         });
   } else {
     // No confirmation necessary for a single block.
-    doDeletion();
+    this.deleteVariableInternal_(name);
   }
 };
 
 /**
+ * Deletes a variable and all of its uses from this workspace. NOTE: Will force
```

Comment:
I'm trying to differentiate it from deleteVariable. If it's clear as-is, I can remove it.